tests/test_combine.py

```diff
@@ -33,6 +33,19 @@ async def test_zip(assert_run):
     xs = stream.zip()
     await assert_run(xs, [])
 
+    # Strict mode (issue #118): Iterable length mismatch raises
+    xs = stream.zip(stream.range(2), stream.range(1), strict=True)
+    with pytest.raises(ValueError):
+        await xs
+
+    # Strict mode (issue #118): No raise for matching-length iterables
+    xs = stream.zip(stream.range(2), stream.range(2), strict=True)
+    await assert_run(xs, [(0, 0), (1, 1)])
+
+    # Strict mode (issue #118): Non-strict mode works as before
+    xs = stream.zip(stream.range(2), stream.range(1))
+    await assert_run(xs, [(0, 0)])
+
```

Comment:
This could probably use another test case that shows that regular exceptions from within the iterables are propagated - do you already have an async iterable that raises an exception for testing somewhere or do we have to build one? :thinking:

Reply:
> do you already have an async iterable that raises an exception for testing

You can use `stream.throw` like here:
https://github.com/vxgmichel/aiostream/blob/5e0547afcebfcb556e011df3c185f69794a187cb/tests/test_combine.py#L161-L165

And you can check for exceptions like this:
https://github.com/vxgmichel/aiostream/blob/5e0547afcebfcb556e011df3c185f69794a187cb/tests/test_combine.py#L114-L117

Reply:
Added two test cases for exception propagation, one for strict and one for non-strict.